### PR TITLE
set annotations on teleport-kube-agent statefulset

### PR DESCRIPTION
**Pull request by seankhliao**
Brings it up to parity with the Deployment variant.
(We want this to use https://github.com/stakater/Reloader ).

changelog: The `teleport-kube-agent` chart now correctly propagates configured annotations when deploying a StatefulSet.

### CODE REVIEW

**Comment by hugoShaka** [comment]
Thank you for your contribution,

To ensure backward compatibility and avoid bugs we require every Helm chart feature to be tested and documented.
The contribution guidelines describe how to add tests and documentation: https://github.com/gravitational/teleport/blob/master/examples/chart/CONTRIBUTING.md

Here you don't need documentation as you don't introduce new values.

Once the PR has tests, please ping us for a second review.

File: examples/chart/teleport-kube-agent/tests/statefulset_test.yaml
Comment (hugoShaka) [inline examples/chart/teleport-kube-agent/tests/statefulset_test.yaml:59]:
```suggestion
```

Nit: the snapshot test is overkill, I know we already have a bunch of them all over the place but they don't add much.

**Review by hugoShaka — APPROVED** [review]
LGTM with minor comment.

File: examples/chart/teleport-kube-agent/tests/statefulset_test.yaml
Comment (seankhliao) [reply examples/chart/teleport-kube-agent/tests/statefulset_test.yaml]: i've removed it

**Comment by seankhliao** [comment]
that should be all the changes i need to make, the unittest passes locally too.

**Comment by strideynet** [comment]
Hey @seankhliao,

Many thanks for your contribution. As some of our CI jobs do not currently support external contributions, I've raised a Buddy PR - https://github.com/gravitational/teleport/pull/46393 

I'll also take over ensuring that this PR is backported to the appropriate release branches for you.

**Pull request closed without merging**
